src/frontend/js/pages/TeacherDashboardContractsLayout/hooks/useCheckContractArchiveExists/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useRef, useState } from 'react';
+import useContractArchive from 'hooks/useContractArchive';
+import { Nullable } from 'types/utils';
+import { CONTRACT_DOWNLOAD_SETTINGS } from 'settings';
+
+export enum CheckContractArchiveStatus {
+  IDDLE = 'iddle',
+  PULLING = 'pulling',
+}
+
+export interface UseCheckContractArchiveExistsProps {
+  contractArchiveId: Nullable<string>;
+}
+
+const useCheckContractArchiveExist = (
+  { contractArchiveId }: UseCheckContractArchiveExistsProps = { contractArchiveId: null },
+) => {
+  const {
+    methods: { check: checkArchiveExist },
+  } = useContractArchive();
+
+  const [status, setStatus] = useState<CheckContractArchiveStatus>(
+    CheckContractArchiveStatus.IDDLE,
+  );
+  const [isContractArchiveExists, setIsContractArchiveExists] = useState(false);
+  const timeoutRef = useRef<NodeJS.Timeout>();
+
+  const doCheckArchiveExists = async (archiveId: string) => {
```

Comment:
I don't know if this is expected but on my side the polling request are fired multiple times at once:

(Look on the right side they seem to be fired by group of 4)
<img width="2557" alt="Capture d’écran 2024-01-12 à 11 32 36" src="https://github.com/openfun/richie/assets/9628870/a5f6e13a-69bc-4f9a-beee-47998bd182ee">